newsfragments/2841.feature.rst

```diff
@@ -0,0 +1,3 @@
+To better reflect the underlying thread handling semantics,
+`trio.to_thread.run_sync` gained a new keyword argument ``abandon_on_cancel``
+that overrides the older (now deprecated) ``cancellable`` flag.
```

Comment:
```suggestion
To better reflect the underlying thread handling semantics,
`trio.to_thread.run_sync` gained a new keyword argument ``abandon_on_cancel``
that can be used instead of the older ``cancellable`` flag (that might get fully deprecated in the future).
```